packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/redshift-cluster.ts

```diff
@@ -0,0 +1,276 @@
+ */
+export interface RedshiftUser {
+  /**
+   * Username for user that has permission to insert records into a Redshift table.
+   */
+  readonly username: string;
+
+  /**
+   * Password for user that has permission to insert records into a Redshift table.
+   *
+   * Do not put passwords in your CDK code directly.
+   *
+   * @default - a Secrets Manager generated password.
+   */
+  readonly password?: cdk.SecretValue;
+
+  /**
+   * KMS key to encrypt the generated secret.
+   *
+   * @default - default master key.
+   */
+  readonly encryptionKey?: kms.IKey;
+}
+
+/**
+ * A column in a Redshift table.
+ */
+export interface RedshiftColumn {
+  /**
+   * The name of the column.
+   */
+  readonly name: string;
+
+  /**
+   * The data type of the column.
+   */
+  readonly dataType: string;
+}
+
+/**
+ * Properties for configuring a Redshift delivery stream destination.
+ */
+export interface RedshiftDestinationProps extends CommonDestinationProps, CommonDestinationS3Props {
+  /**
+   * The cluster user that has INSERT permissions to the desired output table.
+   */
+  readonly user: RedshiftUser;
+
+  /**
+   * The database containing the desired output table.
+   */
+  readonly database: string;
+
+  /**
+   * The table that data should be inserted into.
+   */
+  readonly tableName: string;
+
+  /**
+   * The table columns that the source fields will be loaded into.
+   */
+  readonly tableColumns: RedshiftColumn[];
+
+  /**
+   * The secret that holds Redshift cluster credentials for a user with administrator privileges.
+   *
+   * Used to create the user that Firehose assumes and the table that data is inserted into.
+   *
+   * @default - the master secret is taken from the cluster.
+   */
+  readonly masterSecret?: secretsmanager.ISecret;
+
+  /**
+   * Parameters given to the COPY command that is used to move data from S3 to Redshift.
+   *
+   * @see https://docs.aws.amazon.com/firehose/latest/APIReference/API_CopyCommand.html
+   *
+   * @default '' - no extra parameters are provided to the Redshift COPY command
+   */
+  readonly copyOptions?: string;
+
+  /**
+   * The length of time during which Firehose retries delivery after a failure.
+   *
+   * @default Duration.hours(1)
+   */
+  readonly retryTimeout?: cdk.Duration;
+
+  /**
+   * The intermediate bucket where Firehose will stage your data before COPYing it to the Redshift cluster.
+   *
+   * @default - a bucket will be created for you.
+   */
+  readonly intermediateBucket?: s3.IBucket;
+
+  /**
+   * The role that is attached to the Redshift cluster and will have permissions to access the intermediate bucket.
+   *
+   * If a role is provided, it must be already attached to the cluster, to avoid the 10 role per cluster limit.
+   *
+   * @default - a role will be created for you.
+   */
+  readonly bucketAccessRole?: iam.IRole;
+}
+
+/**
+ * Redshift delivery stream destination.
+ */
+export class RedshiftCluster implements firehose.IDestination {
```

Comment:
I think this should be called `RedshiftTable`. The destination has to be specific to a table, not just a cluster in general.